lib/APPRemoteControl/src/App.cpp

```diff
@@ -384,3 +386,33 @@ void App_statusChannelCallback(const uint8_t* payload, const uint8_t payloadSize
         application->systemStatusCallback(currentStatus->status);
     }
 }
+
+/**
+ * Receives Turtle speed setpoints over SerialMuxProt channel.
+ *
+ * @param[in] payload       Linear and angular speed setpoints in a TurtleSpeed structure.
+ * @param[in] payloadSize   Size of the TurtleSpeed structure.
+ * @param[in] userData      Instance of App class.
+ */
+void App_turtleChannelCallback(const uint8_t* payload, const uint8_t payloadSize, void* userData)
+{
+    (void)userData;
+    if ((nullptr != payload) && (TURTLE_CHANNEL_DLC == payloadSize))
+    {
+        const TurtleSpeed* turtleSpeedData = reinterpret_cast<const TurtleSpeed*>(payload);
+        DifferentialDrive& diffDrive       = DifferentialDrive::getInstance();
+        int16_t            angularSpeed    = static_cast<int16_t>(turtleSpeedData->angular);
+
+        /* Convert to [steps/s] */
+        int16_t centerSpeed = Util::millimetersPerSecondToStepsPerSecond(turtleSpeedData->linearCenter);
+
+        /* Linear speed is set first. Overwrites all speed setpoints. */
+        diffDrive.setLinearSpeed(centerSpeed);
+
+        /* Angular speed is set on-top of the linear speed. Must be called after setLinearSpeed(). */
+        diffDrive.setAngularSpeed(angularSpeed);
+
+        /* Turtle expects no initial data. Can be called without side-effects when no longer in StartupState. */
+        StartupState::getInstance().notifyInitialDataIsSet();
```

Comment:
Why is this necessary? BTW in the driving state the first set speed will be overwritten!

Reply:
It is necessary to get into DrivingState, as the StartUpState waits for initial data such as Position coordinates. This comes from the ConvoyFollower Application in DCS, but its not necessary for the TurtleSim

Reply:
But the ConvoyFollower is a different app, therefore its not necessary, or?

Reply:
Oh. You are right, sorry. Will change this